lib/query-builder.ts

```diff
@@ -0,0 +1,98 @@
+import { ConditionalPick } from 'type-fest'
```

Comment:
It looks to me like the file is not used/required anywhere. Does it fit into this PR?

Reply:
nope - good catch! this is my local laboratory - will remove it 👍